apps/web/src/common/modules/navigations/gnb/modules/gnb-profile/GNBProfile.vue

```diff
@@ -1,6 +1,162 @@
+        label: v, name: k,
+    }))),
+    currencyMenuItems: computed<MenuItem[]>(() => Object.keys(store.state.settings.currencyRates).map((currency) => ({
+        name: currency,
+        label: `${CURRENCY_SYMBOL[currency]}${currency}`,
+    }))),
+    profileMenuRef: null as null|HTMLElement,
+});
+
+const setVisible = (visible: boolean) => {
+    emit('update:visible', visible);
+};
+const openProfileMenu = () => {
+    setVisible(true);
+};
+const hideProfileMenu = () => {
+    if (state.languageMenuVisible) state.languageMenuVisible = false;
+    setVisible(false);
+};
+const setLanguageMenuVisible = (visible: boolean) => {
+    state.languageMenuVisible = visible;
+};
+const setCurrencyMenuVisible = (visible: boolean) => {
+    state.currencyMenuVisible = visible;
+};
+const handleProfileButtonClick = () => {
+    setVisible(!props.visible);
+};
+const handleClickOutsideLanguageMenu = (e: PointerEvent) => {
+    const profileMenuRef = state.profileMenuRef;
+    if (!profileMenuRef) return;
+    const target = e.target as HTMLElement;
+    setLanguageMenuVisible(false);
+    /*
+                v-on-click-outside directive stops click event bubbling.
+                So when this function is called, hideProfileMenu function will never be called which is bound to profileMenuRef's v-on-click-outside directive.
+                The code below closes the profile menu when the user clicks outside the profileMenuRef.
+             */
+    if (!profileMenuRef.contains(target)) hideProfileMenu();
+};
+const handleClickOutsideCurrencyMenu = (e: PointerEvent) => {
+    const profileMenuRef = state.profileMenuRef;
+    if (!profileMenuRef) return;
+    const target = e.target as HTMLElement;
+    setCurrencyMenuVisible(false);
+    if (!profileMenuRef.contains(target)) hideProfileMenu();
+};
+const handleLanguageDropdownClick = () => {
+    setLanguageMenuVisible(!state.languageMenuVisible);
+};
+const handleCurrencyDropdownClick = () => {
+    setCurrencyMenuVisible(!state.currencyMenuVisible);
+};
+
+const handleLanguageClick = async (language) => {
+    try {
+        if (store.state.user.language === language) return;
+
+        await store.dispatch('user/setUser', {
+            language,
+            timezone: state.timezone,
+        });
+
+        setLanguageMenuVisible(false);
+        showSuccessMessage(i18n.t('COMMON.GNB.ACCOUNT.ALT_S_UPDATE'), '');
+    } catch (e) {
+        ErrorHandler.handleRequestError(e, i18n.t('COMMON.GNB.ACCOUNT.ALT_E_UPDATE'));
+    }
+};
+const handleCurrencyClick = async (currency) => {
+    store.commit('settings/setCurrency', currency);
+    setCurrencyMenuVisible(false);
+    showSuccessMessage(i18n.t('COMMON.GNB.ACCOUNT.ALT_S_UPDATE_CURRENCY'), '');
+};
+
+const handleClickSignOut = async () => {
+    const res: Location = {
+        name: AUTH_ROUTE.SIGN_OUT._NAME,
+        query: { nextPath: vm.$route.fullPath },
+    };
+    await vm.$router.push(res);
```

Comment:
suggetion: Since `vm` will be deprecated in Vue 3, it seems like we should use `useRoute`.. @piggggggggy right...?!

Reply:
```suggestion
const route = useRoute();
const router = useRouter();
...
        query: { nextPath: route.fullPath },
    };
    await router.push(res);
```

Reply:
I know that this hook is supported in Vue 3..??